pycaret/internal/pycaret_experiment/supervised_experiment.py

```diff
@@ -4716,18 +4721,20 @@ def finalize_model(
         self.logger.info(f"master_model_container: {len(self.master_model_container)}")
         self.logger.info(f"display_container: {len(self.display_container)}")
 
-        self.logger.info(str(model_final))
+        self.logger.info(str(pipeline_final))
         self.logger.info(
             "finalize_model() successfully completed......................................"
         )
 
         gc.collect()
-        if not model_only:
-            pipeline_final = deepcopy(self.pipeline)
-            pipeline_final.steps.append(["trained_model", model_final])
-            return pipeline_final
 
-        return model_final
+        if model_only:
+            if self._ml_usecase != MLUsecase.TIME_SERIES:
+                return pipeline_final._final_estimator
+            else:
+                return self._get_final_model_from_pipeline(pipeline_final)
```

Comment:
can we make this a method in supervised experiment and call it here as well, so we don't have a special case, just method overriding?